open-weather-connector-demo/src/com/axonivy/connector/openweather/managedbean/ForecastWeatherBean.java

```diff
@@ -0,0 +1,431 @@
+	}
+
+	public String getCurrentWeatherIconCode() {
+		return currentWeatherIconCode;
+	}
+
+	public String getCurrentWeatherDetail() {
+		return currentWeatherDetail;
+	}
+
+	public String getTime12HourName() {
+		return time12HourName;
+	}
+
+	public String getDateName() {
+		return dateName;
+	}
+
+	public String getSearchCityName() {
+		return searchCityName;
+	}
+
+	public void setSearchCityName(String searchCityName) {
+		this.searchCityName = searchCityName;
+	}
+
+	public String getSearchCountryCode() {
+		return searchCountryCode;
+	}
+
+	public void setSearchCountryCode(String searchCountryCode) {
+		this.searchCountryCode = searchCountryCode;
+	}
+
+	public String getSearchStateCode() {
+		return searchStateCode;
+	}
+
+	public void setSearchStateCode(String searchStateCode) {
+		this.searchStateCode = searchStateCode;
+	}
+
+	private void updateDateName() {
+		dateName = DateTimeFormatterUtilities.formatDate(selectedDate);
+	}
+
+	private void updateTime12HourName() {
+		time12HourName = DateTimeFormatterUtilities.formatTime12Hour(selectedTime);
+	}
+
+	public String getCityName() {
+		return cityName;
+	}
+
+	public List<DailyForecastDisplayInfo> getDailyForecastDisplayInfos() {
+		return dailyForecastDisplayInfos;
+	}
+
+	public LineChartModel getTemperatureModel() {
+		return temperatureModel;
+	}
+
+	public String getUnits() {
+		return units;
+	}
+
+	public String getSpeedUnit() {
+		return speedUnit;
+	}
+
+	public BarChartModel getPrecipitationModel() {
+		return precipitationModel;
+	}
+
+	public int getChartWindowSize() {
+		return chartWindowSize;
+	}
+
+	public int getCurrentChartWindowStartX() {
+		return currentChartWindowStartX;
+	}
+
+	public void setCurrentChartWindowStartX(int currentChartWindowStartX) {
+		this.currentChartWindowStartX = currentChartWindowStartX;
+	}
+
+	public int getCurrentChartWindowEndX() {
+		return currentChartWindowEndX;
+	}
+
+	public void setCurrentChartWindowEndX(int currentChartWindowEndX) {
+		this.currentChartWindowEndX = currentChartWindowEndX;
+	}
+
+	public int prepareCurrentChartWindowStartX() {
+		return dailyForecastDisplayInfos.get(selectedDateIndex).getChartWindowStartX();
+	}
+
+	public int prepareCurrentChartWindowEndX() {
+		return dailyForecastDisplayInfos.get(selectedDateIndex).getChartWindowEndX();
+	}
+
+	public void search() {
+		processAndGroupForecastData();
+		setSelectedDateIndex(0);
+		createTemperatureModel();
+		createPrecipitationModel();
+	}
+
+	private void processAndGroupForecastData() {
+		Optional<Forecast> optionalForecast = ForecastService.getInstance()
+				.fetchForecastThreeHourlyFiveDay(searchCityName, searchCountryCode, searchStateCode, units);
+
+		if (optionalForecast.isEmpty()) {
+			return;
+		}
+		Forecast forecast = optionalForecast.get();
+
+		forecast.getList().forEach(record -> {
+			record.getWeather().forEach(weather -> {
+				String icon = weather.getIcon();
+				if (icon != null) {
+					icon = icon.replace('n', 'd');
```

Comment:
Please create static final variables for 'n' and 'd'. Also, comment on why we need to do that.

Reply:
The icon with 'd' in the name is icon for daytime (main color is orange). Otherwise 'n' for night (main color is black). I think the icon color should not be black in my UI, just in my perspective.

Reply:
If changing to variable, could you help me think the name for these variables.

Reply:
ICON_DAY and ICON_NIGHT maybe